src/librbd/ObjectMap.cc

```diff
@@ -134,12 +134,13 @@ bool ObjectMap<I>::object_may_not_exist(uint64_t object_no) const
 
 template <typename I>
 bool ObjectMap<I>::update_required(const ceph::BitVector<2>::Iterator& it,
-                                   uint8_t new_state) {
+                                   uint8_t new_state, bool skip_pending_check) {
   ceph_assert(ceph_mutex_is_locked(m_lock));
   uint8_t state = *it;
   if ((state == new_state) ||
       (new_state == OBJECT_PENDING && state == OBJECT_NONEXISTENT) ||
-      (new_state == OBJECT_NONEXISTENT && state != OBJECT_PENDING)) {
+      (!skip_pending_check && // Deep copy doesn't set state to pending
+       (new_state == OBJECT_NONEXISTENT && state != OBJECT_PENDING))) {
```

Comment:
I am not happy about this change. It looks hackish and I don't like adding yet another param (`skip_pending_check`) to `aio_update`.

Though right now I don't have an idea about a better approach.

Reply:
The only other way I could think of to prevent doing this is to actually use the RBD apis to write/discard etc as those will handle this correctly. To continue with the rados operations, this is the only option I see.
This will also mean that the performance hit is restricted to the mirror daemon io so other ops should be fine.

Reply:
@idryomov , what is your opinion on this?

Reply:
Just thinking, may be we just need to add `force` flag to `aio_update` and skip `update_required` check if it is set (in deep copy)?

Reply:
@trociny , do you mean rename skip_pending_check to force?

Reply:
Rather not rename, but add `force` instead of adding `skip_pending_check`. And it will just skip calling `update_required` if `force` is set. I have not carefully evaluated if it works but if it does then I would prefer this approach instead of adding `skip_pending_check` flag.

Reply:
I'll look into that.

Reply:
Done.